results/code-size.tex

```diff
@@ -0,0 +1,173 @@
+
+By considering the size of the largest and the smallest build for each program we see that the compress factor when applying optimizations is 2.3x on average.
+
+\subsection{Executables}
+
+This subsection consideres the code size of actual runnable programs.
+Functinally equivalent versions are written in both Rust and C and here we compare the size of the binaries produced at each optimization level presented in \autoref{tab:size:settings}.
+For the C program an additional optimization level, -Os, is given as the rightmost bar.
+This level tells the compiler to optimize for code size, it is not available for the Rust compiler at time of writing.
+The first figure, \autoref{fig:size:bin:large}, in this section looks at the size of binaries generated for the two projects described in \autoref{sec:projects}.
+
+\begin{figure}[H]
+  \begin{center}
+    \includegraphics[scale=0.5]{results/plots/size/bin/large/size.png}
+  \end{center}
+  \caption{Code size for Project binaries}
+  \label{fig:size:bin:large}
+\end{figure}
+
+The same trend as for the Libraries is seen for the Rust programs, higher optimizations levels provides a smaller binary.
+Again we see that the O2 level provides a small size gain over the O3 version.
+The compression factor between the larges and smalles binaries is 2.3x on avarage for the Rust code.
+
+The C code does not exibit the same size loss when optimized and the compression factor is 1.3x on average, with Os being the smallest.
+
+\autoref{fig:size:bin:small} shows the code size of a minimal program to boot the Gecko.
+Both implementations only contain an infinite loop and are evaluate to illustate the overhead of the languages.
+
+\begin{figure}[H]
+  \begin{center}
+    \includegraphics[scale=0.5]{results/plots/size/bin/small/size.png}
+  \end{center}
+  \caption{Code size for minimal program}
+  \label{fig:size:bin:small}
+\end{figure}
+
+As explained in \autoref{sec:impl:booting} Rust does not require any aditional initialization or setup compared to C.
+This is evendient in \autoref{fig:size:bin:small} where the binaries are the same size when applying optimizations.
+The debug and unoptimized Rust build is 10x lager than the optimized versions.
+
+\autoref{tab:size:c-vs-rust} compares the size of the smallest C and Rust binaries shown above.
+The number reported is generated by dividing the size of the Rust binary by the size of the C binary giving the factor of witch the Rust version is larger than the C version.
+
+\begin{table}[H]
+  \centering
+  \begin{tabular}{|c|c|c|}
+    \hline
+    Circle Game & Sensor Tracker & Minimal Main \\
+    \hline
+    1.22x & 3.57x & 1.03x \\
+    \hline
+  \end{tabular}
+  \caption{Rust code size relative to C}
+  \label{tab:size:c-vs-rust}
+\end{table}
+
+We see from \autoref{tab:size:c-vs-rust} that the Rust binaries are on average 1.94x larger than the C binaries, although some of the individual programs comes much closer.
```

Comment:
Ja, enig med det. Jeg tok egentlig med det tallet siden det er `~2x` for optimalisert og `~7x` for debug. Men jeg har fortsatt et håp om at vi kan si at det er '~1x' for optimalisert. :|